model/src/main/kotlin/ScanRecord.kt

```diff
@@ -50,6 +50,21 @@ data class ScanRecord(
         @JsonInclude(JsonInclude.Include.NON_EMPTY)
         val data: CustomData = emptyMap()
 ) {
+    /**
+     * Return a de-duplicated map of all errors mapped by [Identifier].
+     */
+    fun collectErrors(): Map<Identifier, Set<OrtIssue>> {
+        val collectedErrors = mutableMapOf<Identifier, MutableSet<OrtIssue>>()
```

Comment:
Nit: Could make this (like in other places) an expression function using

    = mutableMapOf<Identifier, MutableSet<OrtIssue>>().also { collectedErrors -> ... }